### PR TITLE
Add warning when using older version of torch for ViltFeatureExtractor

### LINKED ISSUE #16637: `transformers.ViltProcessor` requires torch >= 1.10
If you follow the instructions in the docs for ViltProcessor:

```python
processor = ViltProcessor.from_pretrained("dandelin/vilt-b32-mlm")
model = ViltModel.from_pretrained("dandelin/vilt-b32-mlm")

inputs = processor(image, text, return_tensors="pt")
outputs = model(**inputs)
last_hidden_states = outputs.last_hidden_state
```

With an older version of torch, you will get:
```
TypeError: meshgrid() got an unexpected keyword argument 'indexing'
```

Maybe there should be a warning when someone tries to import ViltProcessor with an older version of torch?

### PR DESCRIPTION
**Pull request by xhluca**
Closes https://github.com/huggingface/transformers/issues/16637

### CODE REVIEW

**Review by NielsRogge — APPROVED** [review]
LGTM, @LysandreJik can you confirm this is a proper way of handling this?

**Comment by LysandreJik** [comment]
I'd update to use `logging.warning` instead of `warnings.warn`, but other than that it looks like a sound approach.

**Comment by xhlulu** [comment]
@LysandreJik I'm not sure i understand. `transformers.logging.WARN` is a constant.

**Comment by xhlulu** [comment]
Ok I made the change

**Comment by LysandreJik** [comment]
Sorry should have been clearer, in this instance you should move the `logger` instantiation above, and use `logger.warning`. See an example here:

https://github.com/huggingface/transformers/blob/4975002df50c472cbb6f8ac3580e475f570606ab/src/transformers/pipelines/base.py#L234-L237

**Comment by xhlulu** [comment]
@LysandreJik Ok done

**Review by LysandreJik — APPROVED** [review]
LGTM!

**Comment by LysandreJik** [comment]
Ah, before merging we'll need to update the code quality.

Could you just run the code quality tool to ensure that the code quality passes? You can install them with the following, from the root of your clone:
```
pip install -e ".[quality]"
```
And then run them with:
```
make fixup
```

**Comment by xhlulu** [comment]
@LysandreJik just updated it

**Comment by NielsRogge** [comment]
@xhlulu it seems that the CI still isn't green, you can click on the check "check_code_quality" above to see why it's failing.

**Comment by xhlulu** [comment]
```
Traceback (most recent call last):
  File "/home/circleci/.local/bin/doc-builder", line 8, in <module>
    sys.exit(main())
  File "/home/circleci/.local/lib/python3.6/site-packages/doc_builder/commands/doc_builder_cli.py", line 43, in main
    args.func(args)
  File "/home/circleci/.local/lib/python3.6/site-packages/doc_builder/commands/style.py", line 28, in style_command
    raise ValueError(f"{len(changed)} files should be restyled!")
ValueError: 30 files should be restyled!

Exited with code exit status 1

CircleCI received exit code 1
```

**Comment by xhlulu** [comment]
@NielsRogge i just updated to match upstream, should work now

**Comment by NielsRogge** [comment]
Hmm @xhlulu I checked and there's no `meshgrid` being used in `ViltFeatureExtractor`.

It's only the model that requires torch 1.10 or higher, right? Not the feature extractor?

**Comment by xhlulu** [comment]
It's in `ViltEmbeddings`:

https://github.com/huggingface/transformers/blob/39f8eafc1b6f00769240f714e2df5b2c5f111c32/src/transformers/models/vilt/modeling_vilt.py#L115-L151

**Comment by xhlulu** [comment]
I moved the error message, but when trying to run the make fixup, it gives this:

(venv) xhlu@XHL-Desktop:~/dev/transformers$ make fixup
make: execvp: /bin/sh: Argument list too long
make: *** [Makefile:10: modified_only_fixup] Error 127

File: src/transformers/models/vilt/feature_extraction_vilt.py [hunk 1]
```diff
@@ -29,10 +29,10 @@
 )
 from ...utils import TensorType, is_torch_available, logging
 
-
 if is_torch_available():
```
Comment (LysandreJik) [inline src/transformers/models/vilt/feature_extraction_vilt.py:32]:
```suggestion

if is_torch_available():
```

**Review by LysandreJik — APPROVED** [review]
Manually suggested the edit that would make `make fixup` work

**Pull request merged**